source/postprocess/mobility_statistics.cc

```diff
@@ -0,0 +1,201 @@
+                for (unsigned int q=0; q<fe_face_values.n_quadrature_points; ++q)
+                  {
+                    const double vel_mag = velocities[q].norm();
+                    local_sqvel += ((vel_mag * vel_mag) * fe_face_values.JxW(q));
+                    local_fe_face_area += fe_face_values.JxW(q);
+                  }
+                // then merge them with the squared velocities
+                // we found for other faces with the same boundary indicator
+                const types::boundary_id boundary_indicator
+                  = cell->face(f)->boundary_id();
+       
+                local_velocity_square_integral_surface[boundary_indicator] += local_sqvel;
+                local_boundary_area[boundary_indicator] += local_fe_face_area;
+               }
+
+       // now communicate to get the global values
+       std::map<types::boundary_id, double> global_rms_vel;
+       {
+         // first collect local values in the same order in which they are listed
+         // in the set of boundary indicators
+         std::vector<double> local_velocity_square_integral_surface_values;
+         std::vector<double> local_boundary_area_values;  
+
+        for (const auto boundary_id : boundary_indicators)
+          {
+            local_velocity_square_integral_surface_values.push_back (local_velocity_square_integral_surface[boundary_id]);
+            local_boundary_area_values.push_back (local_boundary_area[boundary_id]);
+          }
+        // then collect contributions from all processors
+        std::vector<double> global_velocity_square_integral_values (local_velocity_square_integral_surface_values.size());
+        Utilities::MPI::sum(local_velocity_square_integral_surface_values,this->get_mpi_communicator(),global_velocity_square_integral_values);
+        std::vector<double> global_boundary_area_values (local_boundary_area_values.size());
+        Utilities::MPI::sum(local_boundary_area_values,this->get_mpi_communicator(),global_boundary_area_values);
+
+        // and now take them apart into the global map again
+        // At the same time, calculate the rms velocity for each boundary
+        unsigned int index = 0;
+        for (const auto boundary_id: boundary_indicators)
+          {
+           global_rms_vel[boundary_id] = std::sqrt(global_velocity_square_integral_values[index] / global_boundary_area_values[index]);
+           ++index;
+          }
+       }
+       // now add the computed rms velocities to the statistics object
+       // and create a single string that can be output to the screen
+       //std::ostringstream screen_text;
+       unsigned int index = 0;
+       double mobility =0.0;       
+
+       for (std::map<types::boundary_id, double>::const_iterator
+            svrms = global_rms_vel.begin();
+            svrms != global_rms_vel.end();
+            ++svrms, ++index)
+       
+        {
+           mobility  = svrms[top_boundary_id]/vrms; 
+        }
```

Comment:
This was the problem you mentioned in your message, correct? In a number of places you currently loop over all boundary indicators (because velocity boundary statistics considers all boundaries), while for your purpose you only want to consider a single boundary id (namely `top_boundary_id`). So what you want to do is to remove all the loops over boundary ids (I will mark them) and use `top_boundary_id` as the single boundary indicator.

So here:

```suggestion
           const double mobility  = global_rms_vel[top_boundary_id]/vrms; 
```